### PR TITLE
Add support for no_std

### LINKED ISSUE #205: Nit: std::collections::HashSet
I noticed this library uses `std::collections::HashSet`. Is there a reason for that over `hashbrown::HashSet`? Standard's implementation is backed by hashbrown, so functionally there's no difference. The current code forces a std import.

### PR DESCRIPTION
**Pull request by notgull**
This is a breaking change as it changes the HasContext trait.

Closes #205

### CODE REVIEW

**Comment by grovesNL** [comment]
The CI runners are a bit old so they'll time out here - need to get these fixed

File: Cargo.toml [hunk 1]
```diff
@@ -27,11 +27,15 @@ name = "glow"
 path = "src/lib.rs"
 
 [dependencies]
+ahash = { version = "0.8.3", default-features = false }
```
Comment (grovesNL) [inline Cargo.toml:30]: These should be optional when targeting `std`

**Review by grovesNL — COMMENTED** [review]
Thank you! CI should work again now if you want to try to rebase this

**Comment by notgull** [comment]
Closing for now, as rebasing it would take some work.

**Pull request closed without merging**